setup.py

```diff
@@ -95,12 +95,19 @@ def _get_readme():
                         'passlib>=1.6.2,<2.0',
                         'pexpect>=3.3,<4.0',
                         'requests>=2.4.1,<3.0',
-                        'pytest>=3.0.0,<4.0',
-                        'pytest-catchlog>=1.2.2,<2.0',
-                        'pytest-cov>2.0,<3.0',
-                        'pytest-xdist>=1.14,<2.0',
-                        'pep8>=1.7.0',
-                        'pylint>=1.6.0,<2.0',
-                        'requests_mock>=1.0.0,<2.0',
                         'lxml>=3.6.0,<4.0',
-                        ])
+                        ],
+      extras_require={
+          'tests': [
+                'pytest>=3.0.0,<4.0',
+                'pytest-catchlog>=1.2.2,<2.0',
+                'pytest-cov>2.0,<3.0',
+                'pytest-xdist>=1.14,<2.0',
+                'pep8>=1.7.0',
+                'pylint>=1.6.0,<2.0',
+                'requests_mock>=1.0.0,<2.0',
+          ],
+          'postgres': ['psycopg2>=2.7,<3.0'],
+          'keystone-auth-backend': ['keystonemiddleware>=4.17,<5.0'],
+          'keystone-client': ['python-keystoneclient>=3.13,<4.0'],
```

Comment:
I waffled on that. I'm fine combining tgem if you guys think it makes more sense.

I guess to some extent I think it would make sense to split the client library into a separate package altogether, but that's it's own can of worms.